using_redcap_api/using_redcap_api.md

```diff
@@ -436,6 +436,300 @@ Click on "result" in the environment pane to open a file viewer.
+
+<div class = "warning">
+<b style="color: rgb(var(--color-highlight));">Warning!</b><br>
+
+If you use a system like git to version control your code (which we recommend you do!), then your token is compromised if it has **ever** been committed in a repository that is now public, even if it has since been overwritten.
+
+</div>
+
+
+<div class = "help">
+<b style="color: rgb(var(--color-highlight));">Troubleshooting help</b><br>
+
+What should you do if you think your API token has been compromised?
+
+No matter how brief the exposure was, if your API token was online, you need to discard it and generate a new one.
+See the section above on [generating API tokens](#api-token) for instructions on how to get a new one.
+
+If your REDCap project contains protected information, such as protected health information (PHI), you may need to file a breach report.
+Consult with legal and ethics support at your institution if you're not sure how the law applies to you.
+
+</div>
+
+The example code from the REDCap API playground saves your API token in the first line of the script.
+Although this is okay if that script is saved in a secure way on your computer and will never be shared or published, it's very dangerous if you might end up sharing your code at some point (which we certainly hope you will!).
+
+**So how can you make your API token available to you when you run your script, but not have it saved anywhere where it could accidentally be shared online?**
+
+We'll walk through two strategies, one in Python and one in R.
+Feel free to just read the one in the coding language you prefer.
+
+### Protecting your API token in Python
+
+To protect your API token, never save it in your Python script or Jupyter notebook with the rest of your code.
+Instead, we can use [environment variables](https://developer.vonage.com/en/blog/python-environment-variables-a-primer) to save the API key separately so it's available on your computer but never saved in your Python code.
+
+There are lots of environment variables set for you automatically by your operating system (the [PATH variable](https://en.wikipedia.org/wiki/PATH_(variable) is one you may have encountered before --- if not, no worries).
+You can add any new environment variables you want to and make them available to you in Python by creating a `.env` file.
+
+<div class = "cool-fact">
+<b style="color: rgb(var(--color-highlight));">Did you know?</b><br>
+
+Environment variables are case sensitive on Unix-like systems (e.g. Mac or Linux computers), but not on Windows systems.
+So `redcap_api_token` and `REDCAP_API_TOKEN` could be two different environment variables on a Unix computer, but they would refer to the same thing on a Windows computer!
+
+By convention, environment variables are usually written in all caps, but you can choose to use lower case if you prefer.
+
+</div>
+
+Create the .env file
+---
+
+The `.env` file is just a text file, so you can create it in any text editor you like.
```

Comment:
It's possible it isn't really necessary, but we could potentially add a little something about text editors vs. something like Word (which some folks might think of as a text editor.

Reply:
Oh, very good call! Thank you!